x/auth/ante/unordered.go

```diff
@@ -49,26 +52,43 @@ func (d *UnorderedTxDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, _ bool, ne
 		return next(ctx, tx, false)
 	}
 
+	txHash := sha256.Sum256(ctx.TxBytes())
+
+	// check for duplicates
+	if d.txManager.Contains(txHash) {
+		return ctx, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "tx %X is duplicated")
+	}
+
+	headerInfo := d.env.HeaderService.HeaderInfo(ctx)
+	timeoutTimestamp := unorderedTx.GetTimeoutTimeStamp()
+	if !timeoutTimestamp.IsZero() && timeoutTimestamp.Unix() != 0 {
+		if timeoutTimestamp.Before(headerInfo.Time) {
+			return ctx, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "unordered transaction has a timeout_timestamp that has already passed")
+		}
+		if timeoutTimestamp.After(headerInfo.Time.Add(d.maxTimeoutDuration)) {
+			return ctx, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "unordered tx ttl exceeds %d", d.maxUnOrderedTTL)
+		}
+
+		if d.env.TransactionService.ExecMode(ctx) == transaction.ExecModeFinalize {
+			// a new tx included in the block, add the hash to the unordered tx manager
+			d.txManager.AddTimestamp(txHash, timeoutTimestamp)
+		}
+
+		return next(ctx, tx, false)
+	}
+
 	// TTL is defined as a specific block height at which this tx is no longer valid
 	ttl := unorderedTx.GetTimeoutHeight()
```

Comment:
id propose we remove block height in unordered tx. Looking at both i would say users would always opt for time

Reply:
yeah that would simplified the snapshot logic too

Reply:
removed block height for unordered tx